autoinstall/FusionOS/server_without_GUI/ks.cfg

```diff
@@ -0,0 +1,91 @@
+#version=DEVEL
+# Use graphical install
+graphical
+
+%packages
+@^server-product-environment
+kexec-tools
+#open-vm-tools-desktop
+cloud-init
+java-1.8.0-openjdk-headless
+java-1.8.0-openjdk
+ndctl
+iproute
+rdma-core
+librdmacm-utils
+libibverbs
+libibverbs-utils
+infiniband-diags
+perftest
+%end
+
+# Keyboard layouts
+keyboard --xlayouts='us'
+# System language
+lang en_US.UTF-8
+
+# Network information
+network  --bootproto=dhcp --ipv6=auto --activate
+network  --hostname=localhost.localdomain
+
+# Use CDROM installation media
+cdrom
+# SELinux configuration
+selinux --disabled
+
+# X Window System configuration information
+xconfig  --startxonboot
+# Do not run the Setup Agent on first boot
+firstboot --disable
+# System services
+services --enabled="chronyd"
+services --disabled="firewalld"
+
+ignoredisk --only-use={{ boot_disk_name }}
+autopart
+# Partition clearing information
+clearpart --none --initlabel
+
+# System timezone
+timezone Asia/Shanghai --utc
+
+# Shutdown when the install is finished.
+shutdown
+
+# Root password
+rootpw --iscrypted {{ vm_password_hash }}
+# Add SSH key
+sshkey --username=root "{{ ssh_public_key }}"
+
+{% if new_user is defined and new_user != 'root' %}
+user --name={{ new_user }} --password={{ vm_password_hash }} --groups=root,wheel --iscrypted --gecos="{{ new_user }}"
+sshkey --username={{ new_user }} "{{ ssh_public_key }}"
+{% endif %}
+
+%addon com_redhat_kdump --disable --reserve-mb='128'
+
+%end
+
+%post --interpreter=/usr/bin/bash --log=/root/ks-post.log
+if [ -f /etc/cloud/cloud.cfg ]; then
+    sed -i 's/^disable_root:.*/disable_root: false/' /etc/cloud/cloud.cfg
+    sed -i 's/^ssh_pwauth:.*/ssh_pwauth: yes/' /etc/cloud/cloud.cfg
+fi
+
+# Permit root login via SSH
+sed -i 's/^PermitRootLogin .*/PermitRootLogin yes/' /etc/ssh/sshd_config
+
+# Workaround for setting guest_os_family as RHEL
+if [ -f /etc/FusionOS-release ] && [ ! -f /etc/redhat-release ]; then
+    ln -s /etc/FusionOS-release /etc/redhat-release
+fi
+
+{% if new_user is defined and new_user != 'root' %}
+# Add new user to sudoer
+echo '{{ new_user }} ALL=(ALL) NOPASSWD:ALL' >/etc/sudoers.d/{{ new_user }}
+# Enable autologin
+sed -i '/\[daemon\]/a AutomaticLogin={{ new_user }}' /etc/gdm/custom.conf
+sed -i '/\[daemon\]/a AutomaticLoginEnable=True' /etc/gdm/custom.conf
+{% endif %}
+echo '{{ autoinstall_complete_msg }}' >/dev/ttyS0
+%end
```

Comment:
I am considering whether it is necessary to maintain two kickstart files for FusionOS. After all, it is rare to test RHEL-like guest OS without desktop environment. 2 kickstart files will increase our maintenance and regression testing efforts when we have changes about deploy_vm. Unless it is a must-have test scenario, I suggest we only keep one ks.cfg for FusionOS or any other new RHEL-like OS. We can put the ks.cfg under FusionOS/22/ks.cfg, which can be used by FusionOS 22 or later. When there is a new release which can't use FusionOS/22/ks.cfg, supposing FusionOS 30, we can add a new kickstart like FusionOS/30/ks.cfg at that time.

Reply:
FusionOS-23 doesn't support server_with_GUI.

Reply:
Then it is better to change the path to FusionOS/22/ks.cfg and FusionOS/23/ks.cfg

Reply:
FusionOS-22 works well with both server_with_GUI and server_without_GUI modes.

Reply:
I don't think it conflicts. The unattended install config files are our recommendations. We can't cover all kind of installation varieties. If FusionOS 22 supports GUI, we recommend to use a ks.cfg with GUI installation. Or, you need to describe it more clearly in autoinstall/README.md: Fusion 22 can use server_with_GUI/ks.cfg and server_without_GUI/ks.cfg. But Fusion 23 can only use server_without_GUI/ks.cfg.

Reply:
Fixed in autoinstall/README.md for FusionOS 22 and 23 separately.